pkg/workloadmeta/collectors/podman/podman.go

```diff
@@ -0,0 +1,185 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build podman
```

Comment:
The current conventions seem to disagree, but I think there's no harm in always compiling in podman support (this is very much up to discussion, and I'll just concede if anyone disagrees 😅). In any case, I think we need to add this tag to the `go build` calls somewhere in `tasks/*.py` for it to work.

Reply:
I wasn't sure about this one. I saw that docker and containerd are behind a flag and so I did the same for Podman.

Reply:
It is useful to reduce binary size for some Agent variants (like dogstatsd standalone, Heroku, serverless, etc.)

Reply:
What we've done in AD for instance was to include everything when `!serverless` (to match build tags in some other components IIRC). Maybe we could go that route (or something similar) instead of flagging off specific bits?

Reply:
@vboulineau I'm not sure where to add the `podman` tag in `build_tags.py`.

I see that docker is in `DOGSTATSD_TAGS`, `SECURITY_AGENT_TAGS`, and `TRACE_AGENT_TAGS`. Should I add podman to those 3 plus `LINUX_ONLY_TAGS` ?

Reply:
It should be in all Agents that may use the tagger without relying on `AGENT_TAGS`, so your list is accurate I believe.

Reply:
Added the tag to those constants :+1: